src/System.IO.FileSystem/tests/Base/BaseGetSetAttributes.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO.Tests
+{
+    public abstract class BaseGetSetAttributes : FileSystemTest
+    {
+        protected abstract FileAttributes Get(string path);
+        protected abstract void Set(string path, FileAttributes attributes);
+
+        /// <summary>
+        /// Create an appropriate filesystem object at the given path.
+        /// </summary>
+        protected virtual string CreateItem(string path = null)
+        {
+            path =  path ?? GetTestFilePath();
```

Comment:
This path name is going to be named after this location rather than the location it actually came from in the code.  Isn't that going against one of the purposes of GetTestFilePath, which was to create a file name that could be tied back to the test site?